### DISCUSSION TITLE
Data provider does not refresh when new queries or entries?

### DISCUSSION THREAD

**Discussion by MikeYates** [Budibase/budibase#11870] [Help] [answered]
Is this a bug or its normal? Why does my data provider does not refresh when a new query made or data entry to that data provider . I have to manually refresh the page to see the results.

Using modal or side panel is the same, I have to use navigate to another page so that new queries shows up on the table after entering the form.

**Comment by andz-bb** [comment, marked as answer]
hey @MikeYates 

The Data Provider won't automatically refresh, but there is a _Refresh Data Provider_ action that you can use in situations where you need to reload it. If you have your query set to run when you click a button, you can add that refresh action after the _Execute Query_ action.

<img width="1355" alt="image" src="https://github.com/Budibase/budibase/assets/110921612/b0a772cb-3a91-4066-830b-3866ae76134c">


You can [read more about the Refresh Data Provider action here.](https://docs.budibase.com/docs/data-actions#refresh-data-provider) Hope this helps!
